### DISCUSSION TITLE
(Maybe) a FAQ

### DISCUSSION THREAD

**Discussion by chovanecm** [Zdendaki/RailworksDownloader#26] [Q&A] [answered]
Hi Zdeněk,

may I ask you to elaborate a bit (and perhaps publish it as a FAQ for other users) what the benefits of using your dedicated packaging system are (compared to what Steam Workshop / Marketplace offers)?
It might be obvious but I don't create content and I have always been curious about why Czech content is usually distributed in RWP files instead of being available via Workshop.

Thank you!
Martin

**Comment by JachyHm** [comment, marked as answer]
Hi Martin,
lets start from beginning.
When the game was first published, it wasn't available on Steam, and Steam Workshop was thing yet to be done.
The original game developers created lightweight package manager to pack, distribute and install RWPs- it's part of utilities and it's still distributed with Railworks to this day.
Thing is, this package manager only packs what you manually select and only install already packed files you manually download from somewhere.
Basically you could just download and unpack any archive manualy too - with one slight difference - Utilities will remember which version of package you've installed and will notify you when you are trying to install older/same version again.

Steam Workshop was added later on (just few years back IIRC) and except keeping packages always uptodate did not bring any new features.
Therefore most authors who used Utilities/archives still continued to use them rather than moving their whole work to Workshop.

However main disadvantage of both of these assets distribution methods is that you can not search package by files included in it.
Lets say you know that you are missing file `Assets/CS_addon/Pikku/Stavby/ZastavkaBus`, because you found that by using other 3rd party tools like RW_tools or even ingame.
Try putting this file into Google. Althought it's contained in one of Czech most reuploaded packages in TS ever, you can not find anything, except some forum links where other people are missing the same file:
![searchResult](https://user-images.githubusercontent.com/26261651/138732557-b5825894-7056-41c0-9c51-fb9b703717b2.PNG)

However if you put this exact string or even any part of it, or even the filename only into DLS it will not only find the package with this file included, but you can directly download and install it with just one click, and it will also resolve all other needed dependency packages:
![searchResultDLS](https://user-images.githubusercontent.com/26261651/138732606-87ad4ce1-20d2-4156-92a4-fe4c53c67fb1.PNG)

That's the main reason why DLS was created - to keep database of asset files in individual packages and automatically search and download packages with missing files required by routes, scenarios, etc...
It was never meant to be primary download source, altgough one day it may become one.
Otherwise why not having it all in one place.

Hope this explanation will be enough, if not, don't hesitate to ask more.

**Reply by chovanecm** [reply]
Thanks a lot, Jáchym!
I get the point now and I really appreciate the effort you put into this. I remember back in MSTS times there was a discussion about how  package manager could simplify the problem of installing new addons and resolving dependencies.

Sadly, the feedback from the community was rather negative as some content creators were against distribution of their work through channels other than their own site. Therefore I'm happy to see you try to overcome this, good luck!
